authorization/policy.go

```diff
@@ -44,13 +44,20 @@ func (p ActionResourcePolicy) Authorize(ctx context.Context, userID string, auth
 	}
 
 	if !ok {
+		if resource == nil {
+			return custom_problems.Unauthorized(userID, custom_problems.PolicyViolation{
```

Comment:
Nice catch!

Maybe add a small test in handler_test.go?

Reply:
Added tesst